domains/client/domain-operator/src/tests.rs

```diff
@@ -8800,3 +8800,137 @@ fn proof_to_tx(
         .construct_unsigned_extrinsic(pallet_domains::Call::submit_fraud_proof { fraud_proof })
         .into()
 }
+
+// E2E test: call transporter precompile (0x0800) via EVM tx and observe XDM processing
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_transporter_precompile_transfer_to_consensus_v1_e2e() {
+    const TRANSFER_AMOUNT: u128 = 10_000_000_000_000_000_000u128; // 10 tokens (10*10^18 wei)
+    const FUNDING_AMOUNT: u128 = 100_000_000_000_000_000_000u128; // 100 tokens (100*10^18 wei)
+
+    let precompile_address: H160 = H160::from_low_u64_be(2048);
+
+    // Setup EVM domain and consensus nodes
+    let (_directory, mut ferdie, mut alice) =
+        setup_evm_test_nodes(Sr25519Alice, /*private_evm*/ false, None).await;
+
+    // Run the cross domain gossip message worker
+    ferdie.start_cross_domain_gossip_message_worker();
+    produce_blocks!(ferdie, alice, 2).await.unwrap();
+
+    // Open XDM channel between the consensus chain and the EVM domain
+    open_xdm_channel(&mut ferdie, &mut alice).await;
+
+    // Check minimum_transfer_amount() = runtime minimum transfer amount
+    let minimum_transfer_amount_call_result = alice
+        .client
+        .runtime_api()
+        .call(
+            alice.client.info().best_hash,
+            sp_core::H160::zero(), // from
+            precompile_address,    // to
+            sp_core::hashing::keccak_256(b"minimum_transfer_amount()")[0..4].to_vec(), // data
```

Comment:
Developers will have to get this incantation exactly right, is there a way we can make this easier using a constant, or a function that takes a constant function name?

For example:
```rust
            MINIMUM_TRANSFER_AMOUNT_FN_CODE.to_vec(),
```

Or:
```rust
            eth_fn(MINIMUM_TRANSFER_AMOUNT_FN_NAME).to_vec(),
```

Reply:
Developers will have a much smoother experience using libraries like ethers.js or Alloy, this isn't how most users would interface with the contract.